Add Uplay option for AC1

Kreyren opened the pull request.

qparis commented:
Thank you for the contribution. Have you tested the script?

Kreyren commented:
> Thank you for the contribution. Have you tested the script?

If you prefer rolling release, then i did.

If you prefer stable release, then i did not.

qparis commented:
I don't understand

Kreyren commented:
> I don't understand

Then i did not, will test

Kreyren commented:
Results in:
```
javax.script.ScriptException: <eval>:16:1 Expected comma but found ;
};
 ^ in <eval> at line number 16 at column number 1
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.throwAsScriptException(NashornScriptEngine.java:477)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.compileImpl(NashornScriptEngine.java:544)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.compileImpl(NashornScriptEngine.java:531)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:409)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:162)
	at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
	at org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:50)
	at org.phoenicis.scripts.nashorn.NashornScriptInterpreter.runScript(NashornScriptInterpreter.java:35)
	at org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:35)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: jdk.nashorn.internal.runtime.ParserException: <eval>:16:1 Expected comma but found ;
};
 ^
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.AbstractParser.error(AbstractParser.java:297)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.AbstractParser.error(AbstractParser.java:282)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.objectLiteral(Parser.java:2996)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.primaryExpression(Parser.java:2801)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.memberExpression(Parser.java:3623)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.leftHandSideExpression(Parser.java:3407)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.unaryExpression(Parser.java:4440)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.expression(Parser.java:4620)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.conditionalExpression(Parser.java:4772)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.assignmentExpression(Parser.java:4711)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.variableDeclarationList(Parser.java:1633)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.variableStatement(Parser.java:1532)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.statement(Parser.java:1042)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.sourceElements(Parser.java:909)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.program(Parser.java:844)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.parse(Parser.java:325)
	at jdk.scripting.nashorn/jdk.nashorn.internal.parser.Parser.parse(Parser.java:285)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Context.compile(Context.java:1500)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Context.compileScript(Context.java:1467)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.Context.compileScript(Context.java:750)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.compileImpl(NashornScriptEngine.java:542)
	... 10 more
```

Suspecting it beeing issue with quick script.

Kreyren commented:
Results in 
![image](https://user-images.githubusercontent.com/11302521/58996123-8070fa80-87bc-11e9-9b5a-46a489835f37.png)

Should be probably sandboxed

qparis commented:
Does the game run?

Kreyren commented:
> Does the game run?

> Blocked by: #978

Kreyren commented:
Testing:

![image](https://user-images.githubusercontent.com/11302521/59024774-39b0ee00-8818-11e9-96f4-26f70e1adab1.png)

ImperatorS79 commented:
Please run the game using a terminal (container tab->engine tool->open a terminal) to have the log too.

Kreyren commented:
> Please run the game using a terminal (container tab->engine tool->open a terminal) to have the log too.

Meh using system wine for testing, would use phoenicis if there was an option to get `wine` with preloads easily o.o

ImperatorS79 commented:
> Meh using system wine for testing, would use phoenicis if there was an option to get wine with preloads easily o.o

What does it ever means ? Also, since the script are done for phoenicis, you should test them as a regular phoenicis user.

Kreyren commented:
> > Meh using system wine for testing, would use phoenicis if there was an option to get wine with preloads easily o.o
> 
> What does it ever means ? Also, since the script are done for phoenicis, you should test them as a regular phoenicis user.

For me to test it as phoenicis user i have to use `env LD_LIBRARY_PATH=""something" LD_PRELOAD="something" WINEPREFIX="something" WINEDEBUG="something" /full/path/to/phoenicis/wine /full/path/to/winerprefix` which is annoying -> using system wine instead.

ImperatorS79 commented:
Why do you have to use such a command ?! I never do that.

Kreyren commented:
Trying to add vcrun2005 and d3d10

Kreyren commented:
> Why do you have to use such a command ?! I never do that.

So that it's using phoenicis instead of system?

ImperatorS79 commented:
If you use Container->engine tools->open a terminal, the terminal will already have those var set and you are able to launch wine with `wine`.

Kreyren commented:
> If you use Container->engine tools->open a terminal, the terminal will already have those var set and you are able to launch wine with `wine`.

ah didn't know that o.o ty

Kreyren commented:
@ImperatorS79 i think that this is issue on my end since DXVK is trying to load iGPU, can you test on your end?

ImperatorS79 commented:
No, I cannot.

ImperatorS79 commented:
Please post the wine log and the dxgi and dxvk log if any.

ImperatorS79 commented:
You might need 

> Run your game with VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/radeon_icd.x86_64.json for AMD 64 bit or with VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.json for nvidia.

Kreyren commented:
> You might need
> 
> > Run your game with VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/radeon_icd.x86_64.json for AMD 64 bit or with VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.json for nvidia.

Works now -> DXVK verb update is required

Kreyren commented:
Based on my few mins of gameplay game works no issue with said variable.

Blocked by #982 and Ready to merge after.

plata commented:
Please set this to draft. Obviously, you did not test the script but only the game with system Wine which is irrelevant for this pull request. If the pull request is not opened as draft or at least stating that the script is untested, everybody will normally assume that it can be merged. If we cannot rely on this, we will have to simply close the pull requests to avoid that untested scripts show up in the repository.

This comment of course is applicable for any similar pull request.

Kreyren commented:
> Please set this to draft. Obviously, you did not test the script but only the game with system Wine which is irrelevant for this pull request. If the pull request is not opened as draft or at least stating that the script is untested, everybody will normally assume that it can be merged. If we cannot rely on this, we will have to simply close the pull requests to avoid that untested scripts show up in the repository.
> 
> This comment of course is applicable for any similar pull request.

AC I **script** tested, working, ready to merge.

It was expected from you to merge without test in testing so that they can be pushed further after, but i can make them working now and scripts are made based on my experience with all these games since it doesn't need any workarounds on my system..

>  If we cannot rely on this, we will have to simply close the pull requests to avoid that untested scripts show up in the repository.

97% of your repository are scripts that are not working, fixed Quick steam which still needs fix for starting the installation since users have to perform this manually now and fixed uplay which should work no issue.

Kreyren commented:
or you can give me write access to scripts and watch the magic happend

plata commented:
> Meh using system wine for testing

Sorry if I'm wrong but how does this match with

> AC I script tested, working, ready to merge.

Kreyren commented:
> > Meh using system wine for testing
> 
> Sorry if I'm wrong but how does this match with
> 
> > AC I script tested, working, ready to merge.

using system wine for testing to make phoenicis script to be tested after in phoenicis, because i didn't know that i can export variables from phoenicis since you dont have documentation that informs about it.

plata commented:
So you installed with the script and then tested from command line with system Wine? This information should definitely be part of the pr description. I probably makes sense to mark the script as "testing" because of this.
However, I agree that it's not a general show stopper which prevents us from merging.

Kreyren commented:
> So you installed with the script and then tested from command line with system Wine? This information should definitely be part of the pr description. I probably makes sense to mark the script as "testing" because of this.
> However, I agree that it's not a general show stopper which prevents us from merging.

Tested it **ON PHOENICIS** did research on system wine. 

Depending on different system configuration there might be issues for which end-users can fill new issues since i'm not seting up unreliable VM to test them or making backend for multiple systems to test it.

plata commented on Applications/Games/Assassin's Creed/application.json:
```diff
@@ -1,5 +1,5 @@
+    "name"        : "Assassin's Creed",
     "id"          : "applications.games.assassins_creed",
-    "description" : "Assassin's Creed™ is the next-gen game developed by Ubisoft Montreal that redefines the action genre. While other games claim to be next-gen with impressive graphics and physics, Assassin's Creed merges technology, game design, theme and emotions into a world where you instigate chaos and become a vulnerable, yet powerful, agent of change.<br><br>The setting is 1191 AD. The T ...
+    "description" : "Assassin's Creed is the next-gen game developed by Ubisoft Montreal that redefines the action genre. While other games claim to be next-gen with impressive graphics and physics, Assassin's Creed merges technology, game design, theme and emotions into a world where you instigate chaos and become a vulnerable, yet powerful, agent of change.<br><br>The setting is 1191 AD. The Th ...
```
Why is this changed? As far as I can tell, only the "TM" has been removed. It doesn't make sense that we have to update all the translations just for this.
